cpp/ql/src/Critical/MemoryNeverFreed.ql

```diff
@@ -11,6 +11,9 @@
 
 import MemoryFreed
 
-from Expr alloc
-where isAllocationExpr(alloc) and not allocMayBeFreed(alloc)
+from AllocationExpr alloc
+where
+  alloc.requiresDealloc() and
+  not exists(alloc.(NewOrNewArrayExpr).getPlacementPointer()) and
+  not allocMayBeFreed(alloc)
```

Comment:
Shouldn't `requiresDealloc` be false for a placement-new expression? It's hard to use this library if the caller must know about the special cases of `alloc` that are possible.

Reply:
I considered that, but wasn't sure because we also don't know that a placement new expression _doesn't_ require a dealloc either.  I'd imagine it often does, but sometimes the entire pool is expected to be destroyed instead.

In the interest of avoiding false positives I think I'll make this change...

Reply:
Done.